### PR TITLE
HBASE-26006 Update ref guide about the 2.4.x release line

### PR DESCRIPTION
**Pull request by apurtell**

### CODE REVIEW

File: src/main/asciidoc/_chapters/upgrading.adoc [hunk 1]
```diff
@@ -327,6 +327,11 @@ Quitting...
 The RegionServer Grouping feature has been reimplemented. See section
 <<migrating.rsgroup>> in <<ops_mgt>> for more details.
 
+[[upgrade2.4]]
+=== Upgrade from 2.0.x-2.3.x to 2.4+
+There is no special consideration upgrading to hbase-2.4.x from earlier versions. From 2.2.x, it should be
+rolling upgradeable. From 2.1.x or 2.0.x, you will need to clear the <<upgrade2.2>> hurdle first.
+
```
[the remark this answers is not in the document]
Comment (apurtell) [reply src/main/asciidoc/_chapters/upgrading.adoc:334]: This is a cut and paste from the 2.3 section. I guess both are not quite correct.

File: src/main/asciidoc/_chapters/configuration.adoc [hunk 2]
```diff
@@ -291,7 +291,7 @@ link:https://hadoop.apache.org/cve_list.html[CVEs] so we drop the support in new
 
 [cols="1,4*^.^", options="header"]
 |===
-| | HBase-1.4.x | HBase-1.6.x | HBase-1.7.x | HBase-2.3.x
+| | HBase-1.4.x | HBase-1.6.x | HBase-1.7.x | HBase-2.3.x and HBase-2.4.x
```
Comment (Apache9) [inline src/main/asciidoc/_chapters/configuration.adoc:294]: I think we'd better have a separated column for 2.4.x? IIRC 2.3.x will be EOL soon after the final release of 2.3.7. Correct me if I'm wrong.

File: src/main/asciidoc/_chapters/upgrading.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (Apache9) [reply src/main/asciidoc/_chapters/upgrading.adoc:334]: I do not think this is Andrew's fault. He just copied the content from the 2.3 section. And I think Sean you raised a valid point, I'm not sure whether others considered this situation when reviewing the region procedure store related PRs but at least for me, I didn't considered this one(I'm not sure if I represent myself clearly, I do not mean your point is not important, this is something like a blind point we all missed in the past? Maybe this is more accurate?).

So I think we should modify the 2.3 section too to mention the scenario you described here.

File: src/main/asciidoc/_chapters/configuration.adoc (hunk 2 above)
Comment (apurtell) [reply src/main/asciidoc/_chapters/configuration.adoc:294]: Sure, we can do that

**Comment by apurtell** [comment]
@busbey @Apache9 

I cut and pasted the 2.3 section to 2.4 because I am not aware of any additional consideration you need to take for upgrading from 2.3 to 2.4.

It sounds like we need to update both the 2.3 section and any new section for 2.4 to mention the consequences of having the RegionProcedureStore enabled by default in 2.3 and up. As Sean said:

> to upgrade from 2.0-2.2 to either of 2.3 or 2.4 you have to specially handle the roll out of the upgrade to master roles

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 22s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
||| _ Patch Compile Tests _ |
||| _ Other Tests _ |
|  |   |   2m 23s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3641/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3641 |
| Optional Tests |  |
| uname | Linux c7a543fc33bd 4.15.0-147-generic #151-Ubuntu SMP Fri Jun 18 19:21:19 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 42fe5e5baa |
| Max. process+thread count | 49 (vs. ulimit of 30000) |
| modules | C: . U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3641/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 26s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
||| _ Patch Compile Tests _ |
||| _ Other Tests _ |
|  |   |   1m 26s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3641/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3641 |
| Optional Tests |  |
| uname | Linux 2284286f91dd 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 42fe5e5baa |
| Max. process+thread count | 53 (vs. ulimit of 30000) |
| modules | C: . U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3641/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 25s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
||| _ Patch Compile Tests _ |
||| _ Other Tests _ |
|  |   |   1m 24s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3641/1/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3641 |
| Optional Tests |  |
| uname | Linux cef7acdf4623 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 42fe5e5baa |
| Max. process+thread count | 46 (vs. ulimit of 30000) |
| modules | C: . U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3641/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  7s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m 45s |  master passed  |
| +0 :ok: |  refguide  |   3m 50s |  branch has no errors when building the reference guide. See footer for rendered docs, which you should manually inspect.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m 17s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +0 :ok: |  refguide  |   3m 48s |  patch has no errors when building the reference guide. See footer for rendered docs, which you should manually inspect.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 16s |  The patch does not generate ASF License warnings.  |
|  |   |  19m 24s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3641/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3641 |
| Optional Tests | dupname asflicense refguide |
| uname | Linux 5375a7dde165 4.15.0-147-generic #151-Ubuntu SMP Fri Jun 18 19:21:19 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 42fe5e5baa |
| refguide | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3641/1/artifact/yetus-general-check/output/branch-site/book.html |
| refguide | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3641/1/artifact/yetus-general-check/output/patch-site/book.html |
| Max. process+thread count | 65 (vs. ulimit of 30000) |
| modules | C: . U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3641/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 29s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
||| _ Patch Compile Tests _ |
||| _ Other Tests _ |
|  |   |   1m 32s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3641/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3641 |
| Optional Tests |  |
| uname | Linux 4d1f5d3b1aa2 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 42fe5e5baa |
| Max. process+thread count | 52 (vs. ulimit of 30000) |
| modules | C: . U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3641/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache9** [comment]
Any updates here?

**Comment by Apache9** [comment]
Any updates here @apurtell ? I think we need to add the 2.4.x release line information to our ref guide as now it is the only active branch for 2.x...

**Comment by Apache9** [comment]
Ping @apurtell @ndimiduk @busbey , let's finish this one?

Thanks.

**Comment by Apache9** [comment]
#3989 has already been merged, where we mention the possible problem when upgrading to 2.3.

Will open another PR to for updating ref guide about 2.4.x release line.

Shout if you guys still want to work on this PR.

Thanks.

**Comment by Apache9** [comment]
#3992

**Comment by busbey** [comment]
a belated apology for missing this notification Duo. thanks for taking care of things.

**Pull request closed without merging**